snippet-extractor/src/pool.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2023 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::{io, time::Duration};
+
+pub use deadpool::unmanaged::PoolError;
+use deadpool::{
+    unmanaged::{Object, Pool, PoolConfig},
+    Runtime,
+};
+use derive_more::{Deref, DerefMut};
+use tokio::task::spawn_blocking;
+
+use crate::{Config, Error, SnippetExtractor};
+
+#[derive(Clone)]
+pub struct SnippetExtractorPool {
+    // Hint: Internally `Arc`ed
+    pool: Pool<SnippetExtractor>,
+}
+
+impl SnippetExtractorPool {
+    #[allow(clippy::missing_panics_doc)]
+    pub fn new(config: &Config) -> Result<Self, Error> {
+        let num_cpus = num_cpus::get();
```

Comment:
This should also be part of the configuration.

Reply:
ok we can make it a factor later.

Reply:
I already did so when adding the timeout pool config.